events/testdata/kinesis-firehose-response.json

```diff
@@ -3,12 +3,29 @@
      {
        "data": "SGVsbG8gV29ybGQ=",
        "recordId": "record1",
-       "result": "TRANSFORMED_STATE_OK"
+       "result": "TRANSFORMED_STATE_OK",
+       "metadata": {
+         "partitionKeys": {}
+       }
      },
      {
        "data": "SGVsbG8gV29ybGQ=",
        "recordId": "record2",
-       "result": "TRANSFORMED_STATE_DROPPED"
+       "result": "TRANSFORMED_STATE_DROPPED",
+       "metadata": {
+         "partitionKeys": {}
+       }
+     },
+     {
+       "data": "SGVsbG8gV29ybGQ=",
+       "recordId": "record3",
+       "result": "TransformedStateOk",
+       "metadata": {
+         "partitionKeys": {
+           "iamKey1": "iamValue1",
+           "iamKey2": "iamValue2"
```

Comment:
Can you help me find the related AWS documentation for this feature? I want to understand if `interface{}` is truly the right type for this field vs `map[string]interface{}` or `map[string]string`

Reply:
here is the link to documentation and sample code example written in python -> https://docs.aws.amazon.com/firehose/latest/dev/dynamic-partitioning.html

And here is more information - blog post on firehose dynamic partitioning -> https://aws.amazon.com/blogs/big-data/kinesis-data-firehose-now-supports-dynamic-partitioning-to-amazon-s3/

Reply:
Thanks! It appears to me from those examples that it'd be correct to instead define the partition keys type like this

```
type KinesisFirehoseResponseRecordMetadata struct {
	PartitionKeys map[string]string `json:"partitionKeys"`
}
```

Reply:
ok - let me make that change and test it out and re submit PR

Reply:
@bmoffatt changed PartitionKeys data type to map[string]string - also tested code in sandbox environment - it works